pkg/config/config_test.go

```diff
@@ -236,8 +236,56 @@ func TestLoadProxyEnvOnly(t *testing.T) {
 	Datadog.Set("proxy", nil)
 }
 
+func TestLoadProxyDDPrefixedEnvOnly(t *testing.T) {
+	os.Setenv("DD_HTTP_PROXY", "http_url")
+	os.Setenv("DD_HTTPS_PROXY", "https_url")
+	os.Setenv("DD_NO_PROXY", "a,b,c")
+
+	loadProxyFromEnv()
+
+	proxies := GetProxies()
+	assert.Equal(t,
+		&Proxy{
+			HTTP:    "http_url",
+			HTTPS:   "https_url",
+			NoProxy: []string{"a", "b", "c"}},
+		proxies)
+
+	os.Unsetenv("DD_HTTP_PROXY")
+	os.Unsetenv("DD_HTTPS_PROXY")
+	os.Unsetenv("DD_NO_PROXY")
+	Datadog.Set("proxy", nil)
+}
+
+func TestLoadProxyDDPrefixedEnvPrecedenceOverEnv(t *testing.T) {
+	os.Setenv("DD_HTTP_PROXY", "dd_http_url")
+	os.Setenv("DD_HTTPS_PROXY", "dd_https_url")
+	os.Setenv("DD_NO_PROXY", "a,b,c")
+	os.Setenv("HTTP_PROXY", "env_http_url")
+	os.Setenv("HTTPS_PROXY", "env_https_url")
+	os.Setenv("NO_PROXY", "d,e,f")
+
+	loadProxyFromEnv()
+
+	proxies := GetProxies()
+	assert.Equal(t,
+		&Proxy{
+			HTTP:    "dd_http_url",
+			HTTPS:   "dd_https_url",
+			NoProxy: []string{"a", "b", "c"}},
+		proxies)
+
+	os.Unsetenv("NO_PROXY")
+	os.Unsetenv("HTTPS_PROXY")
+	os.Unsetenv("HTTP_PROXY")
+	os.Unsetenv("DD_HTTP_PROXY")
+	os.Unsetenv("DD_HTTPS_PROXY")
+	os.Unsetenv("DD_NO_PROXY")
+	Datadog.Set("proxy", nil)
+}
+
 func TestLoadProxyEnvAndConf(t *testing.T) {
-	os.Setenv("HTTP", "http_env")
+	os.Setenv("HTTP_PROXY", "http_env")
```

Comment:
(btw, I modified this test because I think it wasn't testing something interesting)